apm-agent-plugins/apm-jmx-plugin/src/test/java/co/elastic/apm/agent/jmx/JmxMetricTrackerTest.java

```diff
@@ -111,13 +116,54 @@ void testRemoveMetric() throws Exception {
         }
     }
 
+    @Test
+    void testMBeanAddedLater() throws Exception {
+        setConfig(JmxMetric.valueOf("object_name[foo:type=Bar] attribute[Baz]"));
+        assertThat(jmxMetricTracker.getRegisteredListeners()).hasSize(1);
+        ObjectName objectName = new ObjectName("foo:type=Bar");
+        ManagementFactory.getPlatformMBeanServer().registerMBean(new TestMetric(), objectName);
```

Comment:
Maybe making MBeanServer instance injectable would definitely help testing without side effects on global state.

Reply:
I agree that this would make it a bit cleaner. But the test setup would also be more complex as the other tests rely on the standard MBeans added in `java.lang.management.ManagementFactory#getPlatformMBeanServer`. Unfortunately, `java.lang.management.ManagementFactory#platformComponents` is private. That makes it impossible to create new `MBeanServer` for each test method which also contains the standard MBeans.